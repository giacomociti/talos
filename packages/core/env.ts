import { create } from '@zazuko/env-node'
import { TalosNsFactory } from './lib/ns.js'

export default create(TalosNsFactory)
