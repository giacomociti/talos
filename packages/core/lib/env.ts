import { create } from '@zazuko/env-node'
import { TalosNsFactory } from './ns.js'

export default create(TalosNsFactory)
