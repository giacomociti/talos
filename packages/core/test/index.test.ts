import path from 'path'
import url from 'url'
import { expect } from 'chai'
import formats from '@rdfjs-elements/formats-pretty'
import type { Dataset } from '@zazuko/env/lib/DatasetExt.js'
import $rdf from '../env.js'
import { fromDirectories } from '../index.js'

const testDir = url.fileURLToPath(new URL('../../../test-resources', import.meta.url))
const ns = $rdf.namespace('https://example.com')

$rdf.formats.import(formats)

describe('@hydrofoil/talos-core', () => {
  describe('fromDirectories', () => {
    let dataset: Dataset

    beforeEach(async () => {
      const dirs = [
        path.resolve(testDir, './resources'),
        path.resolve(testDir, './resources.foo'),
        path.resolve(testDir, './resources.bar'),
      ]
      dataset = await fromDirectories(dirs, ns().value)
    })

    it('merges resources from multiple graph documents', async function () {
      const resource = dataset.match(null, null, null, ns())

      expect(await resource.serialize({ format: 'application/trig' })).to.matchSnapshot(this)
    })

    it('merges resources from dataset and graph documents', async function () {
      const resource = dataset.match(null, null, null, ns('/trig/users/john-doe'))

      expect(await resource.serialize({ format: 'application/trig' })).to.matchSnapshot(this)
    })

    it('merges resources from multiple dataset documents', async function () {
      const resource = dataset.match(null, null, null, ns('/trig/users/jane-doe'))

      expect(await resource.serialize({ format: 'application/trig' })).to.matchSnapshot(this)
    })

    it('marks a resource for "overwrite" by default', () => {
      const match = dataset.match(ns(), $rdf.ns.talos.action, null, $rdf.ns.talos.resources)
      const [{ object: action }, ...more] = match

      expect(action).to.deep.eq($rdf.ns.talos.overwrite)
      expect(more).to.be.empty
    })

    it('marks a resource for "merge" when prefix is used', () => {
      const datasetCore = dataset.match(ns('/project/creta/user.group/admins'), $rdf.ns.talos.action, null, $rdf.ns.talos.resources)
      const [{ object: action }, ...more] = datasetCore

      expect(action).to.deep.eq($rdf.ns.talos.merge)
      expect(more).to.be.empty
    })

    it('uses the last representation when is marked to replace other envs', async function () {
      const resource = dataset.match(null, null, null, ns('/only/one'))

      expect(await resource.serialize({ format: 'application/trig' })).to.matchSnapshot(this)
    })

    it('uses the last representation when is marked in another env', async function () {
      const resource = dataset.match(null, null, null, ns('/only/two'))

      expect(await resource.serialize({ format: 'application/trig' })).to.matchSnapshot(this)
    })

    it('uses the last representation when is marked in a dataset document', async function () {
      const resource = dataset.match(null, null, null, ns('/only/three'))

      expect(await resource.serialize({ format: 'application/trig' })).to.matchSnapshot(this)
    })
  })
})
