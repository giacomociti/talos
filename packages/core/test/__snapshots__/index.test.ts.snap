exports["@hydrofoil/talos-core fromDirectories merges resources from multiple graph documents"] = "\n<https://example.com> {\n\t<https://example.com> a <http://schema.org/Thing> ;\n\t\t<http://schema.org/hasPart> <https://example.com/#part>, _:b1, _:namedPart, _:b6 ;\n\t\t<http://schema.org/name> \"Bar environment\" .\n\n\t<https://example.com/#part> <http://schema.org/minValue> \"10\"^^<http://www.w3.org/2001/XMLSchema#integer> ;\n\t\t<http://schema.org/maxValue> \"100\"^^<http://www.w3.org/2001/XMLSchema#integer> .\n\n\t_:b1 <http://schema.org/name> \"totally anonymous\" .\n\n\t_:namedPart <http://schema.org/name> \"Foo part\", \"Bar part\" .\n\n\t_:b6 <http://schema.org/name> \"totally anonymous\" .\n}\n\n";

exports["@hydrofoil/talos-core fromDirectories merges resources from dataset and graph documents"] = "\n<https://example.com/trig/users/john-doe> {\n\t<https://example.com/trig/users/john-doe> a <http://schema.org/Person>, <http://xmlns.com/foaf/0.1/Person> ;\n\t\t<http://www.w3.org/ns/earl#test> \"trig\" ;\n\t\t<http://xmlns.com/foaf/0.1/name> \"John Doe\" ;\n\t\t<http://schema.org/givenName> \"John\" .\n}\n\n";

exports["@hydrofoil/talos-core fromDirectories merges resources from multiple dataset documents"] = "\n<https://example.com/trig/users/jane-doe> {\n\t<https://example.com/trig/users/jane-doe> a <http://schema.org/Person>, <http://xmlns.com/foaf/0.1/Person> ;\n\t\t<http://www.w3.org/ns/earl#test> \"trig\" ;\n\t\t<http://xmlns.com/foaf/0.1/name> \"Jane Doe\" .\n}\n\n";

exports["@hydrofoil/talos-core fromDirectories uses the last representation when is marked to replace other envs"] = "\n<https://example.com/only/one> {\n\t<https://example.com/only/one> <http://schema.org/name> \"bar\" .\n}\n\n";

exports["@hydrofoil/talos-core fromDirectories uses the last representation when is marked in another env"] = "\n<https://example.com/only/two> {\n\t<https://example.com/only/two> <http://schema.org/name> \"bar\" .\n}\n\n";

exports["@hydrofoil/talos-core fromDirectories uses the last representation when is marked in a dataset document"] = "\n<https://example.com/only/three> {\n\t<https://example.com/only/three> <http://schema.org/name> \"bar\" .\n}\n\n";

exports["@hydrofoil/talos-core fromDirectories adds data with SPARQL"] = "\n<https://example.com/trig/users/john-doe-additional> {\n\t<https://example.com/trig/users/john-doe> a <http://xmlns.com/foaf/0.1/Person> ;\n\t\t<http://schema.org/givenName> \"John\" .\n}\n\n";

